examples/examples/event_callback.rs

```diff
@@ -64,8 +55,26 @@ async fn main() {
 						},
 					}
 				},
-				_ => debug!("ignoring unsupported module event: {:?}", evr.event),
+				RuntimeEvent::System(system_event) => {
+					println!(">>>>>>>>>> system event: {:?}", system_event);
+					match &system_event {
+						frame_system::Event::ExtrinsicSuccess { dispatch_info } => {
+							println!("DispatchInfo: {:?}", dispatch_info);
+							return
+						},
+						_ => {
+							debug!("ignoring unsupported system event");
+						},
+					}
+				},
+				_ => debug!("ignoring unsupported module event: {:?}", event_record.event),
 			}
 		}
 	}
+
+	// After we finished whatever we wanted, unusubscribe from the subscription,
+	// to ensure, that the node does not keep sending us events.
+	// This is not mandatory. If the subscription goes out of scope, it should
+	// automatically unsubscribe from  the node.
+	subscription.unsubscribe().unwrap();
```

Comment:
I don't quite see/understand how the automatic `unsubscribe()` works. Don't we need to implement `Drop` for that to work?

Reply:
This is true, but maybe there exists a blanket implementation for that? I could imagine so, but it should be confirmed that this is the case.

Reply:
Not sure what you're asking. So as the subscritpion is from the rpc-client, we basically have three different (un)subscription types. Sidenote: sadly, `drop()`  can not call `unsubscribe` due to unsubscribe requiring `self` and drop only having a reference.

For our two basic clients, the subscription is a simple channel with a receiver and waiting for the sender to send something. 

- Tungstenite: No real unsubscription anyway:  
https://github.com/scs/substrate-api-client/blob/74993c6567a74e329ae3de1dbfd7c2c9cb1636f8/src/rpc/tungstenite_client/subscription.rs#L47-L52

- ws-client:  unsubscribe and drop: https://github.com/scs/substrate-api-client/blob/74993c6567a74e329ae3de1dbfd7c2c9cb1636f8/src/rpc/ws_client/subscription.rs#L50-L62.

- jsonrpsee, for which the whole subscription trait has been designed:  handles everything internally anyway: https://github.com/scs/substrate-api-client/blob/74993c6567a74e329ae3de1dbfd7c2c9cb1636f8/src/rpc/jsonrpsee_client/subscription.rs#L32-L34 ,
resp: https://github.com/paritytech/jsonrpsee/blob/master/core/src/client/mod.rs#L218-L232


Does that answer the questions?

Reply:
Yes, but this basically means that the comment is wrong:

``
	// This is not mandatory. If the subscription goes out of scope, it should
	// automatically unsubscribe from the node.
```

Reply:
Right. I'll correct that.

Reply:
Removed it - better now?

Reply:
Beautiful.